### PR TITLE
New package gcc-linaro-aarch64-linux-gnu

### PR DESCRIPTION
**Pull request by Raybuntu**
New package gcc-linaro-aarch64-linux-gnu that will be used to build kernel packages when doing 64/32bit builds.
Moving gcc-linaro-aarch64-elf and gcc-linaro-arm-eabi to projects/Odroid_C2.

It will require some testing so please don't merge right away. @Kwiboo can you try building your Rockchip projects with gcc-linaro-aarch64-linux-gnu 7.1?
@lrusak As discussed on Slack this might fix your dragonboard issues, thanks for the idea.

### CODE REVIEW

**Comment by Raybuntu** [comment]
I'm moving gcc-linaro-arm-eabi to projects too cause it's only used by Odroid_C2.

**Comment by lrusak** [comment]
thanks @Raybuntu !!!

**Comment by Raybuntu** [comment]
I've made 3 clean ARM builds with this PR over night for C2, WH and WP2. It's building without issues and a quick test on WH and WP2 was ok too.

**Comment by Raybuntu** [comment]
Thanks @Kwiboo

**Comment by Raybuntu** [comment]
Please do not merge yet. There is a small issue with 64/32bit when building u-boot or linux before starting a complete build. cmake.conf will be wrong. That issue is not introduced by this PR but I wanna include a fix with this PR to catch 2 birds with one stone.

File: packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE=""
+PKG_URL="https://releases.linaro.org/components/toolchain/binaries/7.1-2017.08/aarch64-linux-gnu/gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu.tar.xz"
+PKG_SOURCE_DIR="gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu"
+PKG_SECTION="lang"
+PKG_SHORTDESC=""
+PKG_LONGDESC=""
+
+PKG_IS_ADDON="no"
```
Comment (MilhouseVH) [inline packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk]: Four lines above can be dropped (see #2117). Also licence copyright should be updated. A shortdesc would be nice!

**Comment by Raybuntu** [comment]
@MilhouseVH Thanks for the review. I've applied everything and rebased.

File: packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+PKG_VERSION="7.1.1-2017.08"
+PKG_SHA256="cc61203bb1bab9b51e9631dce803be7c00b814577fd7a118113a83448bf6a833"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE=""
+PKG_URL="https://releases.linaro.org/components/toolchain/binaries/7.1-2017.08/aarch64-linux-gnu/gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu.tar.xz"
+PKG_SOURCE_DIR="gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu"
+PKG_SECTION="lang"
+PKG_AUTORECONF="no"
+PKG_SHORTDESC="Linaro Aarch64 GNU Linux Binary Toolchain"
```
Comment (CvH) [inline packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk]: Autoreconf should be after shortdesc

**Comment by Raybuntu** [comment]
@CvH Thanks. Fixed!

File: packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+PKG_NAME="gcc-linaro-aarch64-linux-gnu"
+PKG_VERSION="7.1.1-2017.08"
+PKG_SHA256="cc61203bb1bab9b51e9631dce803be7c00b814577fd7a118113a83448bf6a833"
+PKG_ARCH="any"
+PKG_LICENSE="GPL"
+PKG_SITE=""
+PKG_URL="https://releases.linaro.org/components/toolchain/binaries/7.1-2017.08/aarch64-linux-gnu/gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu.tar.xz"
```
Comment (MilhouseVH) [inline packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk]:
```
-+PKG_URL="https://releases.linaro.org/components/toolchain/binaries/7.1-2017.08/aarch64-linux-gnu/gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu.tar.xz"
++PKG_URL="https://releases.linaro.org/components/toolchain/binaries/${PKG_VERSION}/aarch64-linux-gnu/gcc-linaro-${PKG_VERSION}-x86_64_aarch64-linux-gnu.tar.xz" 
```

?

Comment (Raybuntu) [reply packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk]: @MilhouseVH unfortunately it's 7.1 not 7.1.1

Comment (MilhouseVH) [reply packages/lang/gcc-linaro-aarch64-linux-gnu/package.mk]: Ah!

**Comment by Raybuntu** [comment]
Anything else? Nit picks? Merge?

**Pull request merged**